sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,207 @@
+        /// <param name="permissions">Specifies the permissions granted to this entry.</param>
+        /// <param name="defaultScope">Indicates whether this is the default entry for the ACL.</param>
+        /// <param name="entityId">Specifies the entity for which this entry applies.</param>
+        public PathAccessControlEntry(
+            AccessControlType accessControlType,
+            RolePermissions permissions,
+            bool defaultScope = false,
+            string entityId = null)
+        {
+            DefaultScope = defaultScope;
+            AccessControlType = accessControlType;
+            EntityId = entityId;
+            Permissions = permissions;
+        }
+
+        /// <summary>
+        /// Override of ToString()
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            if (DefaultScope)
+            {
+                stringBuilder.Append("default:");
+            }
+            stringBuilder.Append(AccessControlType.ToString().ToLowerInvariant());
+            stringBuilder.Append(":");
+            stringBuilder.Append(EntityId ?? "");
+            stringBuilder.Append(":");
+            stringBuilder.Append(Permissions.ToSymbolicString());
+
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Parses the provided string into a <see cref="PathAccessControlEntry"/>
+        /// </summary>
+        /// <param name="str">The string representation of the ACL.</param>
+        /// <returns>A <see cref="PathAccessControlEntry"/>.</returns>
+        public static PathAccessControlEntry Parse(string str)
+        {
+            PathAccessControlEntry entry = new PathAccessControlEntry();
+            string[] parts = str.Split(':');
+            int indexOffset = 0;
+
+            if (parts.Length == 4)
+            {
+                if (!parts[0].Equals("default", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    throw Errors.InvalidArgument(nameof(str));
+                }
+                entry.DefaultScope = true;
+                indexOffset = 1;
+            }
+            entry.AccessControlType = ParseAccesControlType(parts[indexOffset]);
+
+            if (!string.IsNullOrEmpty(parts[1 + indexOffset]))
+            {
+                entry.EntityId = parts[1 + indexOffset];
+            }
+
+            entry.Permissions = RolePermissionsExtensions.ParseSymbolic(parts[2 + indexOffset], false);
+            return entry;
+        }
+
+        /// <summary>
+        /// Converts the Access Control List to a <see cref="string"/>.
+        /// </summary>
+        /// <param name="acl">The Access Control List to serialize</param>
+        /// <returns>string</returns>
+        public static string SerializeList(IList<PathAccessControlEntry> acl)
```

Comment:
add null check for acl; also change acl to accessControlList

Reply:
fixed